server/sonar-server/src/main/java/org/sonar/server/usertoken/ws/CreateAction.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.usertoken.ws;
+
+import com.google.common.base.Optional;
+import java.util.Date;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.utils.System2;
+import org.sonar.core.permission.GlobalPermissions;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.user.UserTokenDto;
+import org.sonar.server.exceptions.BadRequestException;
+import org.sonar.server.token.TokenGenerator;
+import org.sonar.server.user.UserSession;
+import org.sonar.server.user.index.UserIndex;
+import org.sonarqube.ws.WsUserTokens.CreateWsResponse;
+import org.sonarqube.ws.client.usertoken.CreateWsRequest;
+
+import static java.lang.String.format;
+import static org.sonar.api.utils.DateUtils.formatDateTime;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+import static org.sonarqube.ws.client.usertoken.UserTokensWsParameters.PARAM_LOGIN;
+import static org.sonarqube.ws.client.usertoken.UserTokensWsParameters.PARAM_NAME;
+
+public class CreateAction implements UserTokensWsAction {
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final System2 system;
+  private final TokenGenerator tokenGenerator;
+  private final UserIndex userIndex;
+
+  public CreateAction(UserSession userSession, DbClient dbClient, System2 system, TokenGenerator tokenGenerator, UserIndex userIndex) {
+    this.userSession = userSession;
+    this.dbClient = dbClient;
+    this.system = system;
+    this.tokenGenerator = tokenGenerator;
+    this.userIndex = userIndex;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction("create")
```

Comment:
Feedback about terminology - "generate" could be more appropriate than "create", as the token value is not chosen by caller but by WS.